autoscaler/azure.py

```diff
@@ -122,6 +180,118 @@ def _get_azure_class(type_):
     return m.group('class')
 
 
+_SCALE_SET_SIZE_LIMIT = 100
+
+
+# Appears as an unbounded scale set. Currently, Azure Scale Sets have a limit of 100 hosts.
+class AzureVirtualScaleSet(AutoScalingGroup):
+    provider = 'azure'
+
+    def __init__(self, region, resource_group, client, instance_type, scale_sets, kube_nodes):
+        self.client = client
+        self.instance_type = instance_type
+        # TODO: Remove this. Legacy value indicating that this node is optimized for compute
+        self.tags = {'openai/computing': 'true'}
+        self.name = 'virtual_scale_set_' + instance_type + '_' + region + '_' + resource_group
+        self.scale_sets = dict((scale_set.name, scale_set) for scale_set in scale_sets)
+        self.desired_capacity = sum(scale_set.sku.capacity for scale_set in scale_sets)
+
+        self.region = region
+        self.resource_group = resource_group
+
+        self.selectors = dict(self.tags)
+        # HACK: for matching node selectors
+        self.selectors['azure/type'] = self.instance_type
+        self.selectors['azure/region'] = self.region
+        self.selectors['azure/class'] = _get_azure_class(self.instance_type)
+
+        self.min_size = 0
+        self.max_size = 1000
```

Comment:
Yes, well, actually I just discovered that Terraform doesn't support Managed Disks for Scale Sets, so we can only have 40 VMs per Scale Set (I'll change the constant) :( So we'd need 25. I just copied this over from the old Azure integration, which had a limit of 1000